### PR TITLE
Add templates and consistency checks

### PR DESCRIPTION
**Pull request by h-mayorquin**
This adds a couple of things:
* A function to erase explicit templates that we have found to be faulty.
* Logic to get consistent shapes in all the templates. For example, because of rounding errors not al the templates had the same temporal samples. We correct for this and also add a check.
* Added a tqdm progress bar to the consolidate script.

### CODE REVIEW

File: python/upload_templates.py [hunk 1]
```diff
@@ -247,6 +259,12 @@ def find_channels_with_max_peak_to_peak_vectorized(templates):
         templates_extension = analyzer.get_extension("templates")
         templates_extension_data = templates_extension.get_data(outputs="Templates")
 
+        # Do a check for the expected shape of the templates
+        number_of_units = sorting.get_num_units()
+        expected_shape = (number_of_units, 240, 384)
```
Comment (alejoe91) [inline python/upload_templates.py]:
```suggestion
        expected_shape = (number_of_units, target_nbefore + target_nafter, 384)
```
easier to read :)

**Review by alejoe91 — COMMENTED** [review]
thanks man! one tiny tiny change for readibility!

File: python/upload_templates.py (hunk 1 above)
Comment (h-mayorquin) [reply python/upload_templates.py]: Yes, same for the number of channels right? Let me correct that.

Comment (h-mayorquin) [reply python/upload_templates.py]: Done in the last commit. I also forgot to stage the files to delete templates : P

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**